### PR TITLE
`Development`: Increase client test coverage of various chart related components

### PR DESCRIPTION
**Pull request by willergeorg**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [ ] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context

This PR increases the line coverage of several chart related components to above 90%. The table below marks the corresponding components and the new coverage.

#### Code Review
- [x] Review 1
- [x] Review 2

### Test Coverage

image

### CODE REVIEW

File: src/test/javascript/spec/component/exam/exam-scores/exam-scores-average-scores-graph.component.spec.ts [hunk 1]
```diff
@@ -105,4 +114,44 @@ describe('ExamScoresAverageScoresGraphComponent', () => {
...
+
+        it('should not navigate if exercise id is missing', () => {
+            component.lookup['test'] = { exerciseType: ExerciseType.QUIZ };
+
+            component.onSelect(event);
+
+            expect(navigateToExerciseMock).toHaveBeenCalledTimes(0);
+        });
+
+        it('should navigate if exercise type is missing', () => {
```
Comment (manuelmanso) [inline src/test/javascript/spec/component/exam/exam-scores/exam-scores-average-scores-graph.component.spec.ts:137]: Shouldn't this be 'should **not** navigate if exercise type is missing' ?

**Review by manuelmanso — DISMISSED** [review]
Nice job 👍

File: src/test/javascript/spec/component/exam/exam-scores/exam-scores-average-scores-graph.component.spec.ts (hunk 1 above)
Comment (willergeorg) [reply src/test/javascript/spec/component/exam/exam-scores/exam-scores-average-scores-graph.component.spec.ts]: Good catch. Fixed. :)

File: src/test/javascript/spec/component/exam/exam-scores/exam-scores-average-scores-graph.component.spec.ts [hunk 2]
```diff
@@ -105,4 +114,44 @@ describe('ExamScoresAverageScoresGraphComponent', () => {
...
+            expect(navigateToExerciseMock).toHaveBeenCalledOnce();
+            expect(navigateToExerciseMock).toHaveBeenCalledWith(42, ExerciseType.QUIZ);
+        });
+
+        it('should not navigate if exercise id is missing', () => {
+            component.lookup['test'] = { exerciseType: ExerciseType.QUIZ };
+
+            component.onSelect(event);
+
+            expect(navigateToExerciseMock).toHaveBeenCalledTimes(0);
```
Comment (Dominik-Weinzierl) [inline src/test/javascript/spec/component/exam/exam-scores/exam-scores-average-scores-graph.component.spec.ts:134]:
```suggestion
            expect(navigateToExerciseMock).not.toHaveBeenCalled();
```

File: src/test/javascript/spec/component/exam/exam-scores/exam-scores-average-scores-graph.component.spec.ts [hunk 3]
```diff
@@ -105,4 +114,44 @@ describe('ExamScoresAverageScoresGraphComponent', () => {
...
+        });
+
+        it('should not navigate if exercise type is missing', () => {
+            component.lookup['test'] = { exerciseId: 42 };
+
+            component.onSelect(event);
+
+            expect(navigateToExerciseMock).toHaveBeenCalledTimes(0);
```
Comment (Dominik-Weinzierl) [inline src/test/javascript/spec/component/exam/exam-scores/exam-scores-average-scores-graph.component.spec.ts:142]:
```suggestion
            expect(navigateToExerciseMock).not.toHaveBeenCalled();
```

**Review by Dominik-Weinzierl — COMMENTED** [review]
Some minor suggestions :+1: Looks good :+1:

**Review by ge65cer — APPROVED** [review]
Tests are always a welcomed addition. 🙂

**Review by manuelmanso — APPROVED** [review]
Reapprove!

**Pull request merged**